### PR TITLE
Refactoring for sourcefile 'common.c'

### PR DESCRIPTION
**Pull request by hydroconstructor**
File divided to some parts. Functions with "flash" in names extracted to common_flash.c, with "option" extracted to option.c etc.
Removed unnecessary headers.
Removed one single function which was used nowhere.
And so on.
Project built under Windows and Ubuntu and seems to be working.

### CODE REVIEW

**Comment by hydroconstructor** [comment]
Problem with some defines... going to fix tomorrow or so.

**Comment by Nightwalker-87** [comment]
@hydroconstructor Please note that there is a PR #1216 already open, so please ensure that
1) there is no conflict with branch `chipid_cleanup`,
2) the build test pass and
3) all features remain functional.

**Comment by hydroconstructor** [comment]
In #1212 pointed:

rewrite and modularisation of: chipid.c chipid.h common.c stlink.h stm32.h

Here, I tried to rewrite and modularise common.c - splitting it to some files.
chipid_cleanup branch has common.c as single file, not divided to some parts.
Does it mean that splitting common.c not necessary right now?

Or maybe split common.c in different branch, not chipid_cleanup - and merge them later?

**Comment by Nightwalker-87** [comment]
As long as it does not conflict with HEAD in branch `chipid_cleanup` and is an addition to #1216 it's ok.
Please avoid/resolve the current conflicts as they will not disappear in any case and must be resolved anyway.

File: CMakeLists.txt [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # General cmake settings
 ###
 
-cmake_minimum_required(VERSION 3.10.2)
+cmake_minimum_required(VERSION 3.7.2)
```
Comment (Nightwalker-87) [inline CMakeLists.txt:5]: Please restore the previous version. This has just been altered previously along with updated os-support.

File: src/calculate.h [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/*
+ * File: calculate.h
+ *
+ * TODO: add a description
```
Comment (Nightwalker-87) [inline src/calculate.h:4]: Yes... :wink:

File: src/option.c [hunk 3]
```diff
@@ -0,0 +1,1027 @@
+#include <stdio.h>
```
Comment (Nightwalker-87) [inline src/option.c:1]: Please name this source file "option_bytes.c" to have a clearer definition.

File: src/map_file.h [hunk 4]
```diff
@@ -0,0 +1,28 @@
+/*
+ * File: map_file.h
+ *
+ * TODO: add a description
```
Comment (Nightwalker-87) [inline src/map_file.h:4]: Yes... :wink:

File: src/common_flash.h [hunk 5]
```diff
@@ -0,0 +1,27 @@
+/*
+ * File: common_flash.h
+ *
+ * TODO: add a description
```
Comment (Nightwalker-87) [inline src/common_flash.h:4]: Yes... :wink:

File: src/common.h [hunk 6]
```diff
@@ -0,0 +1,15 @@
+/*
+ * File: common.h
+ *
+ * TODO: add a description
```
Comment (Nightwalker-87) [inline src/common.h:4]: Yes... :wink:

**Comment by Nightwalker-87** [comment]
@hydroconstructor Could you also address #1155 and #1194 when touching this source file? Please also check if there are any other open issues that can be addressed here without much special efforts. Things appear much more easy to overlook now.

File: CMakeLists.txt (hunk 1 above)
Comment (Nightwalker-87) [reply CMakeLists.txt]: Fixed in commit d98d3a50edbbf984b1fe842908e1420d26c3ff0f.

File: src/calculate.h (hunk 2 above)
Comment (Nightwalker-87) [reply src/calculate.h]: Fixed in commit d98d3a50edbbf984b1fe842908e1420d26c3ff0f.

File: src/option.c (hunk 3 above)
Comment (Nightwalker-87) [reply src/option.c]: Fixed in commit d98d3a50edbbf984b1fe842908e1420d26c3ff0f.

File: src/map_file.h (hunk 4 above)
Comment (Nightwalker-87) [reply src/map_file.h]: Fixed in commit d98d3a50edbbf984b1fe842908e1420d26c3ff0f.

File: src/common_flash.h (hunk 5 above)
Comment (Nightwalker-87) [reply src/common_flash.h]: Fixed in commit d98d3a50edbbf984b1fe842908e1420d26c3ff0f.

File: src/common.h (hunk 6 above)
Comment (Nightwalker-87) [reply src/common.h]: Fixed in commit d98d3a50edbbf984b1fe842908e1420d26c3ff0f.

**Review by Nightwalker-87 — APPROVED** [review]

**Pull request merged**